components/user-mgt/org.wso2.carbon.identity.user.action/src/main/java/org/wso2/carbon/identity/user/action/internal/listener/ActionUserOperationEventListener.java

```diff
@@ -93,9 +93,9 @@ public boolean doPreUpdateCredentialByAdminWithID(String userName, Object creden
                 return true;
             } else if (executionStatus.getStatus() == ActionExecutionStatus.Status.FAILED) {
                 Failure failure = (Failure) executionStatus.getResponse();
-                String errorMsg = buildErrorMessage(failure.getFailureReason(), failure.getFailureDescription());
-                throw new UserStoreClientException(errorMsg,
-                        UserActionError.PRE_UPDATE_PASSWORD_ACTION_EXECUTION_FAILED);
+                throw new UserActionExecutionClientException(
+                        UserActionError.PRE_UPDATE_PASSWORD_ACTION_EXECUTION_FAILED,
+                        failure.getFailureReason(), failure.getFailureDescription());
```

Comment:
Let's throw `UserActionExecutionServerException` for below UserStoreException as well.

And lets replace all `UserStoreException `

Reply:
The decision was only client exceptions, as we want to pass the both failure message and description.

Reply:
Since we have a specific exception type for user action execution, isn't it better to throw that without throwing the generic exception.

We can just add the server exceptions and no need to specifically handle them in the scim or user core layer.